### PR TITLE
implement not found page

### PR DESCRIPTION
**Pull request by DmytroHryshyn**
## What does this PR do?

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/app/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const NotFound = async () => {
+  const h = headers();
+
+  const nonce = h.get("x-nonce") ?? undefined;
+
+  const { i18n } = await getProps();
+
+  return (
+    // @ts-expect-error withTrpc expects AppProps
+    <PageWrapper requiresLicense={false} pageProps={{ i18n }} nonce={nonce} themeBasis={null} i18n={i18n}>
```
Comment (hbjORbj) [inline apps/web/app/not-found.tsx:27]: Let's put this into `not-found/layout.tsx` and this current file into `not-found/page.tsx` for consistency.

Comment (hbjORbj) [reply apps/web/app/not-found.tsx:27]: not relevant for this page. Please ignore.

**Comment by hbjORbj** [comment]
i think we need to revert the changes in yarn.lock?

File: apps/web/app/not-found.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      <NotFoundPage />
+    </PageWrapper>
+  );
+};
+
+export default NotFound;
+export const runtime = "nodejs";
```
[the remark this answers is not in the document]
Comment (DmytroHryshyn) [reply apps/web/app/not-found.tsx:31]: I've added that to fix a bug on next 13.4.6, its not needed anymore. as well as additional env settings

[the remark this answers is not in the document]
Comment (DmytroHryshyn) [reply apps/web/app/not-found.tsx:31]: (maybe it was never needed)

**Review by grzpab — APPROVED** [review]

**Pull request merged**